influxdb3_catalog/src/catalog.rs

```diff
@@ -579,14 +602,27 @@ impl DatabaseSchema {
             })
     }
 
-    pub fn table_definition(&self, table_name: impl Into<Arc<str>>) -> Option<&TableDefinition> {
+    pub fn table_definition(
+        &self,
+        table_name: impl Into<Arc<str>>,
+    ) -> Option<Arc<TableDefinition>> {
         self.table_map
             .get_by_right(&table_name.into())
-            .and_then(|table_id| self.tables.get(table_id))
+            .and_then(|table_id| self.tables.get(table_id).cloned())
     }
 
-    pub fn table_definition_by_id(&self, table_id: TableId) -> Option<&TableDefinition> {
-        self.tables.get(&table_id)
+    pub fn table_definition_by_id(&self, table_id: TableId) -> Option<Arc<TableDefinition>> {
+        self.tables.get(&table_id).cloned()
+    }
+
+    pub fn table_definition_and_id(
+        &self,
+        table_name: impl Into<Arc<str>>,
+    ) -> Option<(TableId, Arc<TableDefinition>)> {
```

Comment:
Shouldn't the `TableDefinition` have the ID? Then you just return that?

Reply:
Yeah, it does, this API is sort of unnecessary in that sense, but there are a lot of places that use both so I added this method. There are similar methods for getting `DatabaseSchema` from the catalog and `ColumnDefinition`s from the table.